text/0039-burn-redemption-memo.md

```diff
@@ -0,0 +1,75 @@
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+This is a simple proposal. We propose reserving the memo type 0x0001 for the burn redemption memo:
+
+| Memo type bytes | Name                                              |
+| -----------     | -----------                                       |
+| 0x0001          | Burn redemption memo                              |
+
+
+## 0x0001 Burn redemption memo
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 64     | Unstructured payload  |
+
+The payload of the burn redemption memo is left unstructured to give burners/bridge operators/etc the flexibility of storing whatever information they desire in order to associate the burn with activity outside of the MobileCoin blockchain. For example, this could contain an ERC20 address of the receipient of the tokens been handed out as a result of the burn, a transaction hash linking the burn to the transaction on the external blockchain that corresponds to the burn on the MobileCoin blockchain, etc.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In our opinion, there is no reason why not to reserve a memo type for this purpose. We are very far from running out of memo types, and having this memo would enable the use-cases of linking burn transactions to outside activity as mentioned above.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Given that we want a way to include information such as a correlated ERC20 address or transaction id on an external blockchain with a MobileCoin burn transaction, using RTH is the natural way to do so without needing a ledger format change.
+
+While it is possible to add more information to the ledger to store this kind of data, it would make no sense as the majority of `TxOut`s are not burn `TxOut`s. Additionally, the memo field was created exactly for this kind of auxiliary data.
+
+We could enforce a schema over the payload, e.g. the first byte could specify what kind of address or id is stored in the remaining 63 bytes. However, this seems suboptimal since it is impossible for us to predict how this field would get used and we do not want to enforce arbitrarily limits on entities choosing to use this mechanism to track burns. Additionally, there is no way for Consensus to actually verify the contents of the memo - so such schema would be a recommendation at most, but anyone using this memo could still store whatever they desire in the payload and ignore the schema.
+
+We could avoid standardizing this memo field altogether, but as mentioned above, this is not enforceable in the transaction validation code. As such, we think it is better to explicitly reserve a memo type for this, so that it does not get re-purposed by future MCIPs or properly-behaved clients.
```

Comment:
IMHO, it definitely is better and we might consider reserving a small bit of extra space for the burn memos. They could potentially fall under "minting memos"